src/mod/server.mod/servmsg.c

```diff
@@ -1075,15 +1118,25 @@ static void server_activity(int idx, char *msg, int len)
   code = newsplit(&msg);
   if (raw_log && ((strcmp(code, "PRIVMSG") && strcmp(code, "NOTICE")) ||
       !match_ignore(from))) {
-    if (!strcmp(from, ""))
-      putlog(LOG_RAW, "*", "[@] %s %s", code, msg);
-    else
-      putlog(LOG_RAW, "*", "[@] %s %s %s", from, code, msg);
+      rawlen = egg_snprintf(s, sizeof s, "[@] ");
+      if (tag) {
+        rawlen += egg_snprintf(s + rawlen, sizeof s - rawlen, "%s ", tag);
+      }
+      if (strcmp(from, "")) {
+        rawlen += egg_snprintf(s + rawlen, sizeof s - rawlen, "%s ", from);
+      }
+      egg_snprintf(s + rawlen, sizeof s - rawlen, "%s %s", code, msg);
+      putlog(LOG_RAW, "*", "%s", s);
```

Comment:
where did "from" go in the printing if it's not ""?
I'd suggest just printing it in msgtag format `[@] nick!ident@host @a=b PRIVMSG #chan :hi`
or whatever else makes sense here

Reply:
Does this make more sense?